src/components/cache/managers/redis-cache.manager.ts

```diff
@@ -1,26 +1,29 @@
 import { ICacheManager, SetOptions } from './cache.manager.interface';
-import { PackageUtils } from '../utils/package-loader';
-import { IRedisCacheOptions } from './types';
-import Logger from '../components/logger';
+import { PackageUtils } from '../../../utils/package-loader';
+import Logger from '../../logger';
 
-export class RedisCacheManager<T> implements ICacheManager<T> {
+export interface IRedisOptions {
+  url: string;
+}
+
+export class RedisCacheManager implements ICacheManager {
   private redisManager: any;
 
-  constructor(options: IRedisCacheOptions) {
+  constructor(options: IRedisOptions) {
     const { createClient } = PackageUtils.loadPackage('redis') as any;
     this.redisManager = createClient(options);
     this.redisManager.connect().catch((e) => Logger.error('Failed to connect to redis', e));
   }
 
-  public async set(key: string, data: T, options?: SetOptions): Promise<void> {
```

Comment:
why is the generic type moved to here instead of the class definition?
will it not break clients?

Reply:
it won't, it's actually less restrictive:
- when you put `<T>` to the class ,then you enforce `<T>` to every method call ever
- when you put `<T>` to the method, then you can pass there two different types of param in two consecutive calls